internal/driver/driver.go

```diff
@@ -0,0 +1,729 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2023 YIQISOFT
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// This package provides an example implementation of
+// S7 interface.
+package driver
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/device-sdk-go/v3/pkg/interfaces"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v3/pkg/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/models"
+	"github.com/robinson/gos7"
+	"github.com/spf13/cast"
+)
+
+const readCommandsExecutedName = "ReadCommandsExecuted"
+
+const (
+	// Word Length
+	s7wlbit     = 0x01 //Bit (inside a word)
+	s7wlbyte    = 0x02 //Byte (8 bit)
+	s7wlChar    = 0x03
+	s7wlword    = 0x04 //Word (16 bit)
+	s7wlint     = 0x05
+	s7wldword   = 0x06 //Double Word (32 bit)
+	s7wldint    = 0x07
+	s7wlreal    = 0x08 //Real (32 bit float)
+	s7wlcounter = 0x1C //Counter (16 bit)
+	s7wltimer   = 0x1D //Timer (16 bit)
+)
+
+var once sync.Once
+var driver *Driver
+
+type Driver struct {
+	lc          logger.LoggingClient
+	asyncCh     chan<- *sdkModel.AsyncValues
+	stringArray []string
```

Comment:
do we still need this variable?

Reply:
I don't known. Copied from sample code. 😄